### PR TITLE
Designed a Logo for RPL and Update Docs

### PR DESCRIPTION
**Pull request by stuuupidcat**
- Designed a logo for RPL 🥳 (Ferris with a magnifying glass 🦀 🔍).
- Add .rpl files for debugged pattern files.
- Released an rpl-language-support VSCode extension and welcome contributions to the [repository](https://github.com/stuuupidcat/rpl-language-support).

### CODE REVIEW

File: docs/patterns/CVE-2020-35892-3-MIR.rpl [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+patt {
+    use core::ops::range::Range<usize>;
+
+    // how to express constraints on meta variables?
+    // 1. use a where clause (how to define a where clause?)
+    // 2. use a predicate (how to define a predicate?)
+    p[
+        $T: ty,
+        $SlabT: ty
+    ] where $SlabT: Adt
```
Comment (stuuupidcat) [inline docs/patterns/CVE-2020-35892-3-MIR.rpl:12]: Here is a question about RPL design:
How to express constraints on meta variables?
1. use a where clause (how to define a where clause?)
2. use a predicate (how to define a predicate?)
@frank-king @TheVeryDarkness

File: docs/patterns/CVE-2018-20992-MIR.rpl [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    use std::vec::Vec;
+
+    p_vec_set_len_to_extend[
+        $T: ty
+    ] = #[mir] pub fn _ (..) -> _ {
+        let $vec: Vec<$T> = _;
+        let $new_len: usize = _;
+        let $vec_ref: &Vec<$T> = &$vec;
+        let $old_len: usize = copy ((*$vec_ref).len);
+        // SwitchInt Here or If here?
```
Comment (stuuupidcat) [inline docs/patterns/CVE-2018-20992-MIR.rpl:13]: If I try to express `#without!(if condition) {}` in RPL MIR pattern, should I use `#without! if` or `#without! switchInt`?
@frank-king @TheVeryDarkness

File: docs/patterns/CVE-2020-35892-3-MIR.rpl (hunk 1 above)
Comment (frank-king) [reply docs/patterns/CVE-2020-35892-3-MIR.rpl:12]: If the meta variable has some constraints, I think we should define a new pattern instead.

File: docs/patterns/CVE-2018-20992-MIR.rpl (hunk 2 above)
Comment (frank-king) [reply docs/patterns/CVE-2018-20992-MIR.rpl:13]: I think `if` can be added as a syntactic sugar for `switchInt` with a boolean operand.

**Pull request merged**